test/base/src/androidTest/kotlin/io/realm/test/shared/RealmInstantTests.kt

```diff
@@ -0,0 +1,138 @@
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .path("$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    // Test both unmanaged and managed boundaries
+    @Test
+    fun timestamp_boundaries() {
+        roundTrip(RealmInstant(Long.MIN_VALUE, -999_999_999)) { min ->
+            assertEquals(Long.MIN_VALUE, min.epochSeconds)
+            assertEquals(-999_999_999, min.nanosecondsOfSecond)
+            assertEquals(RealmInstant.MIN, min)
+        }
+
+        roundTrip(RealmInstant(Long.MAX_VALUE, 999_999_999)) { max ->
+            assertEquals(Long.MAX_VALUE, max.epochSeconds)
+            assertEquals(999_999_999, max.nanosecondsOfSecond)
+            assertEquals(RealmInstant.MAX, max)
+        }
+
+        roundTrip(RealmInstant(Long.MAX_VALUE, Int.MAX_VALUE)) { maxOverflow ->
+            assertEquals(RealmInstant.MAX, maxOverflow)
+        }
+
+        roundTrip(RealmInstant(Long.MAX_VALUE, 1_000_000_000)) { minOverflow ->
+            assertEquals(RealmInstant.MAX, minOverflow)
+        }
+
+        roundTrip(RealmInstant(Long.MIN_VALUE, Int.MIN_VALUE)) { maxUnderflow ->
+            assertEquals(RealmInstant.MIN, maxUnderflow)
+        }
+
+        roundTrip(RealmInstant(Long.MIN_VALUE, -1_000_000_000)) { minUnderflow ->
+            assertEquals(RealmInstant.MIN, minUnderflow)
+        }
+
+        roundTrip(RealmInstant(0, 0)) { zero ->
+            assertEquals(0, zero.epochSeconds)
+            assertEquals(0, zero.nanosecondsOfSecond)
+        }
+
+        roundTrip(RealmInstant(0, 1)) { zeroPlusOne ->
+            assertEquals(0, zeroPlusOne.epochSeconds)
+            assertEquals(1, zeroPlusOne.nanosecondsOfSecond)
+        }
+
+        roundTrip(RealmInstant(0, -1)) { zeroMinusOne ->
+            assertEquals(0, zeroMinusOne.epochSeconds)
+            assertEquals(-1, zeroMinusOne.nanosecondsOfSecond)
+        }
+
+        roundTrip(RealmInstant(-1, 2_000_000_200)) { crossingZeroFromNegative ->
+            assertEquals(RealmInstant(1, 200), crossingZeroFromNegative)
+        }
+
+        roundTrip(RealmInstant(1, -2_000_000_200)) { crossingZeroFromPositive ->
+            assertEquals(RealmInstant(-1, -200), crossingZeroFromPositive)
+        }
+    }
+
+    // Store value and retrieve it again
+    private fun roundTrip(timestamp: RealmInstant, function: (RealmInstant) -> Unit) {
+
+        // Test unmanaged objects
+        function(timestamp)
+
+        // Test managed objects
+        realm.writeBlocking {
+            val sample = copyToRealm(Sample())
+            sample.timestampField = timestamp
+            function(sample.timestampField)
```

Comment:
I would prefer that we asserted on a value retrieved from the realm. If we mess up something in the accessors the value could technically survive in the backing field. Pulling the sample through a query would ensure that it has indeed been round tripping the realm. I guess the `SampleTest` ensures that the accessors are actually directing to the realm, but still more comforting if we also did it here.

Reply:
Sure, I can rewrite the test.